File: secondary_implementations/python/modules/LaposteModule.py
import requests
from lxml import html
from AbstractModule import AbstractModule

class LaposteModule(AbstractModule):
    def try_login(self, uname, password):
        agent = AbstractModule().user_agent()
        headers = {'user-agent': agent}
        resp = requests.get("https://www.laposte.net/accueil", headers=headers)
        tree = html.fromstring(resp.content)
        payload = tree.forms[0].fields
        payload = {k: v for k, v in payload.iteritems()}
        payload['login'] = uname
        payload['password'] = password
        headers = {'user-agent': agent, 'Content-Type': 'application/x-www-form-urlencoded'}
        loginResp = requests.post("https://compte.laposte.net/login.do", data=payload, headers=headers,
                                  cookies=resp.cookies, allow_redirects=False)

        if loginResp.status_code == 302:
            return True
        else:
            return False


laposte = LaposteModule()

File: src/main/scala/fish/philwants/modules/LaposteModule.scala
package fish.philwants.modules

import fish.philwants.Credentials
import scala.collection.JavaConversions._
import org.jsoup.nodes.FormElement
import fish.philwants.JsoupImplicits._

object LaposteModule extends AbstractModule {
  val uri = "https://laposte.net/"
  val moduleName = "Laposte"

  def tryLogin(creds: Credentials): Boolean = {
    val loginUri = "https://www.laposte.net/accueil"
    val resp = get(loginUri)
      .validateTLSCertificates(false)
      .execute()

    // Get the form and update it with credentials
    val form = resp.firstForm
      .update("login", creds.username)
      .update("password", creds.password)

    // Try to login
    val loginResp = form
      .submit()
      .header("Content-Type", "application/x-www-form-urlencoded")
      .cookies(resp.cookies())
      .followRedirects(false)
      .validateTLSCertificates(false)
      .execute()

    // Check login result
    loginResp.statusCode() == 302
  }
}
